### ISSUE TITLE
How to update EMA only at desired steps

### ISSUE THREAD

**Issue by dohyun1411** [lucidrains/vector-quantize-pytorch#221]
Hi,

I am training my model (VQ + other models) with gradient accumulation.
Since the VQ module uses EMA updates at every forward pass,
it is difficult to align them with the gradient updates.

What I want is to accumulate the necessary data during forward passes,
but defer the actual EMA update until a specific step (e.g., when gradients are applied).

I also noticed a parameter, manual_ema_update.
Can this be used to manually trigger EMA updates at the right time?

**Comment by lucidrains** [comment]
@dohyun1411 hey Dohyun, do you want to see if [this](https://github.com/lucidrains/vector-quantize-pytorch/commit/0fa6e78c7d20612f79bc7e3d11252380114190e4#diff-d9cee953279014ae3c9536028cc2d6245448a0853870d4ab091568dbc6c91961R438) works for you on the latest version?
